libr/bin/dwarf.c

```diff
@@ -330,27 +262,34 @@ static const ut8 *r_bin_dwarf_parse_lnp_header(
 				break;
 			}
 			buf += len + 1;
-			if (buf>=buf_end) { buf = NULL; goto beach; }
-			buf = r_uleb128 (buf, buf_end-buf, &id_idx);
-			if (buf>=buf_end) { buf = NULL; goto beach; }
-			buf = r_uleb128 (buf, buf_end-buf, &mod_time);
-			if (buf>=buf_end) { buf = NULL; goto beach; }
-			buf = r_uleb128 (buf, buf_end-buf, &file_len);
-			if (buf>=buf_end) { buf = NULL; goto beach; }
+			if (buf >= buf_end) {
```

Comment:
This could be simplified with a macro in the future @HoundThe, in the subsequent pull requests.